js/src/hooks/useCreateAccounts.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useCreateMCAccount from '../components/google-mc-account-card/useCreateMCAccount';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+import useExistingGoogleMCAccounts from '.~/hooks/useExistingGoogleMCAccounts';
+import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+
+/**
+ * Custom hook to handle the creation of MC and Ads accounts.
+ */
+const useCreateAccounts = () => {
+	/**
+	 * Refs are used to avoid the re-render of the parent component.
+	 *
+	 * accountCreationChecksResolvedRef - Indicates if the account creation checks have been resolved.
+	 * isCreatingAccountsRef - Indicates if the accounts are being created.
+	 * accountsCreatedRef - Indicates if the accounts have been created.
+	 */
+	const accountCreationChecksResolvedRef = useRef( false );
+	const isCreatingAccountsRef = useRef( false );
+	const accountsCreatedRef = useRef( false );
+
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasFinishedResolutionForExistingAdsccount,
+	} = useGoogleAdsAccount();
+
+	const {
+		data: existingMCAccounts,
+		hasFinishedResolution: hasFinishedResolutionForExistingMCAccounts,
+	} = useExistingGoogleMCAccounts();
+
+	const {
+		existingAccounts: existingAdsAccount,
+		isResolving: isResolvingExistingAdsAccount,
```

Comment:
@ankitrox Let's use `hasFinishedResolution` instead. This means updating the `useExistingGoogleAdsAccounts` hook to add the property. The main reason is `isResolving` initially returns `false` and then `true`. This may lead to some conditions not being met. `hasFinishedResolution` will be `false` initially and then `true` when it's over.

Reply:
`useExistingGoogleAdsAccounts` hook does not return `hasFinishedResolution`. If we want to use that we will have to do 

```
const hasFinishedResolution = select( STORE_KEY ). hasFinishedResolution( 'getExistingGoogleAdsAccounts', [] );
```

which seems unnecessary because under the hood both `isResolving` and `hasFinishedResolution` uses `getResolutionState` as can be seen [here for hasFinishedResolution](https://github.com/WordPress/gutenberg/blob/0ed82aec5f97a6e5fc06d0654f645865d8484caa/packages/data/src/redux-store/metadata/selectors.js#L90-L93) and [here for isResolving](https://github.com/WordPress/gutenberg/blob/0ed82aec5f97a6e5fc06d0654f645865d8484caa/packages/data/src/redux-store/metadata/selectors.js#L135-L139).

Reply:
That's fine with updating the hook @ankitrox by adding the `hasFinishedResolution`. True they use `getResolutionState` under the hood but check my original comment about their initial values and you can test. `isResolving` will return `true` only when it's triggered. Meaning when the component is mounted and between the trigger, it'll return `false`. So it might lead to unwanted behaviours. That's why we prefer to use `hasFinishedResolution` since it's `false` initially until resolved.